semanticcpg/src/main/scala/io/shiftleft/semanticcpg/accesspath/TrackingPointToAccessPath.scala

```diff
@@ -9,101 +9,180 @@ import org.slf4j.LoggerFactory
+      case _: nodes.Unknown               => (TrackedUnknown, Nil)
+      // FieldIdentifiers are only fake arguments, hence should not be tracked
+      case _: nodes.FieldIdentifier => (TrackedUnknown, Nil)
+      case block: nodes.Block =>
+        TrackingPointMethodsBase
+          .lastExpressionInBlock(block)
+          .map { toTrackedBaseAndAccessPathInternal }
+          .getOrElse((TrackedUnknown, Nil))
+      case call: nodes.Call if !MemberAccess.isGenericMemberAccessName(call.name) => (TrackedReturnValue(call), Nil)
+
+      case memberAccess: nodes.Call =>
+        //assume: MemberAccess.isGenericMemberAccessName(call.name)
+        memberAccess.name match {
+          case Operators.memberAccess | Operators.indirectMemberAccess =>
+            if (!hasWarnedDeprecations) {
+              logger.warn(s"Deprecated Operator ${memberAccess.name} on ${memberAccess}")
+              hasWarnedDeprecations = true
             }
-          case None =>
-            logger.warn(
-              s"Invalid AST: Found member access without second argument." +
-                s" Member access CODE: ${memberAccess.code}" +
-                s" In method ${memberAccess.method.fullName}")
-            Elements.normalized(List(VariableAccess))
+            memberAccess
+              .argumentOption(1)
+              .map { toTrackedBaseAndAccessPathInternal }
+              .map {
+                case (base, tail) =>
+                  (base,
+                   memberAccess
+                     .argumentOption(2)
+                     .collect {
+                       case lit: nodes.Literal   => ConstantAccess(lit.code)
+                       case id: nodes.Identifier => ConstantAccess(id.name)
+                     }
+                     .getOrElse(VariableAccess) :: tail)
+              }
+              .getOrElse {
+                logger.warn(s"Missing argument on call ${memberAccess}.")
+                (TrackedUnknown, Nil)
+              }
+
+          case Operators.computedMemberAccess | Operators.indirectComputedMemberAccess =>
+            if (!hasWarnedDeprecations) {
+              logger.warn(s"Deprecated Operator ${memberAccess.name} on ${memberAccess}")
+              hasWarnedDeprecations = true
+            }
+            memberAccess
+              .argumentOption(1)
+              .map { toTrackedBaseAndAccessPathInternal }
+              .map {
+                case (base, tail) =>
+                  (base,
+                   memberAccess
+                     .argumentOption(2)
+                     .collect {
+                       case lit: nodes.Literal => ConstantAccess(lit.code)
+                     }
+                     .getOrElse(VariableAccess) :: tail)
+              }
+              .getOrElse {
+                logger.warn(s"Missing argument on call ${memberAccess}.")
+                (TrackedUnknown, Nil)
+              }
+          case Operators.indirection =>
```

Comment:
Looks like there's a lot of duplication here and the function is very long. Any way we can reduce the amount of code here?